### ISSUE TITLE
Chat - App does not display user name of sender until visited in reply in thread

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#26252]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app and login with user A
2. Open user B report
3. Send any image to user B
4. Open the app in another device and login with user B
5. Open user A report and click on reply in thread on hover on image sent in step 3
6. Send any image to reply in thread using user B
7. In user A device, observe the new reply in thread chat in LHN, it does not display username of user B besides '[Attachment]'
8. Open the reply in thread report and observe that now it displays user B username besides '[Attachment]'

## Expected Result:
App should display username of user who has sent the message in LHN

## Actual Result:
App does not display username of user in LHN if user has sent Attachment in reply in thread

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.58-3

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856 

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/d1669558-3fde-4cdb-b3ff-e8d357a3e98b

https://github.com/Expensify/App/assets/115492554/ddb2cde3-d0fe-4362-9002-07ad4b9b9ec5

**Expensify/Expensify Issue URL:**

**Issue reported by:** @dhanashree-sawant

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692702613182719

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by yh-0218** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App does not display user name of sender until visited in reply in thread.
In the case of message, This error happens.
### What is the root cause of that problem?
The cause of problem is that the length of `participantPersonalDetailList` is 1 before click thread.
https://github.com/Expensify/App/blob/4675db5348e305daca320f91066476344191d8d2/src/libs/SidebarUtils.js#L267
### What changes do you think we should make in order to solve the problem?
We can add thread case `|| result.isThread` there.
https://github.com/Expensify/App/blob/4675db5348e305daca320f91066476344191d8d2/src/libs/SidebarUtils.js#L267
```
const hasMultipleParticipants = participantPersonalDetailList.length > 1 || result.isChatRoom || result.isPolicyExpenseChat || result.isThread;
```
### What alternative solutions did you explore? (Optional)
We can make `hasMultipleParticipants` as `true` when `result.isThread && participantPersonalDetailList.length > 0`

**Comment by alexpensify** [comment]
I will review soon

**Comment by alexpensify** [comment]
@jasperhuangg, @narefyev91, @dukenv0307 - I believe this is a dupe of https://github.com/Expensify/App/issues/21334. Can one of y'all confirm and we can close this one out? Thanks!

**Comment by dhanashree-sawant** [comment]
Hi @alexpensify, #21334 is deployed a month back on live, the following issue is still occuring on latest staging

**Comment by jasperhuangg** [comment]
@alexpensify I don't believe this is a duplicate issue, it seems like it touches a different flow.

**Comment by alexpensify** [comment]
Thanks, everyone for the feedback, marking this as `external`. 

@parasharrajat when you get a chance, can you review the proposal? Thanks!

**Comment by alexpensify** [comment]
@parasharrajat any update here? Thanks!

**Comment by alexpensify** [comment]
@parasharrajat any update if you were able to review the proposal here?

**Comment by yh-0218** [comment]
@parasharrajat Thanks for you reply. This is my opinion
### What is the root cause of that problem?
User-A attach one image or send a message.
User-B reply to that image or message(attack image to User-A's image, reply a message to User-A's message)
Then User-A can see new channel for User-B's reply.
For this channel, `participantPersonalDetailList` length is 1 (only User-B because User-A didn't join that channel yet.)
When User-A click reply channel, then `participantPersonalDetailList` length is 2(User-A and User-B).
So I thought 2 way to fix this issue. [Proposal](https://github.com/Expensify/App/issues/26252#issuecomment-1698970623)

**Comment by alexpensify** [comment]
@parasharrajat any feedback for the new and updated proposals?

**Comment by alexpensify** [comment]
@parasharrajat - any update here?

**Comment by alexpensify** [comment]
@parasharrajat any update to share?

**Comment by alexpensify** [comment]
@parasharrajat any news here?

**Comment by alexpensify** [comment]
@parasharrajat could you please review @dukenv0307's feedback when you get a chance? Thanks!

**Comment by alexpensify** [comment]
Thanks!

**Comment by alexpensify** [comment]
@parasharrajat any update here?

**Comment by alexpensify** [comment]
@parasharrajat - if you are unable to review, please let me know. I'm going to need to reassign it soon and keep it moving forward. I am not sure if you have notes outside of this GH that are not shared yet. Thanks for the update!

**Comment by alexpensify** [comment]
No update

**Comment by alexpensify** [comment]
I've requested help here: https://expensify.slack.com/archives/C02NK2DQWUX/p1696949291510719

**Comment by alexpensify** [comment]
I'm reassigning to @situchan. Please review the proposals, and we can keep this one moving forward.

**Comment by situchan** [comment]
This happens because BE doesn't send pusher event of updated participants when user joins the thread. I think it's intentional but engineer will confirm.

As I don't see any harm treating thread as chat room, I am fine with this solution in FE.
@yh-0218's [proposal](https://github.com/Expensify/App/issues/26252#issuecomment-1698970623) LGTM!

I am also open to send pusher event in backend with updated participants when user joins the thread but not recommended as this is kind of "dynamic updating" issue and also for performance.

🎀 👀 🎀 C+ reviewed

**Comment by iwiznia** [comment]
I am not sure I follow...
> The cause of problem is that the length of participantPersonalDetailList is 1 before click thread.

Then we should fix this, no?

> We can add thread case || result.isThread there.

The fact that it's a thread does not mean it has multiple participants, does it? Like I can start a thread in a message I sent and that thread would just have 1 participant (just me).

**Comment by situchan** [comment]
> The fact that it's a thread does not mean it has multiple participants, does it? Like I can start a thread in a message I sent and that thread would just have 1 participant (just me).

Exactly.
The real bug here is that `participantAccountIDs` in my Onyx is not updated even after I joined the thread already and another user sends message on this thread, as this video clearly shows:


https://github.com/Expensify/App/assets/108292595/49bf893d-a1e7-49f1-9136-a19b27ee01a7

**Comment by iwiznia** [comment]
ok, then I think we need to fix that. I assume that problem is in the server, as it is not sending the correct pusher update?

**Comment by situchan** [comment]
> ok, then I think we need to fix that. I assume that problem is in the server, as it is not sending the correct pusher update?

I think so

**Comment by alexpensify** [comment]
@iwiznia - do we need to edit the proposal or take it internal?

**Comment by alexpensify** [comment]
@iwiznia - bumping again, we need your feedback on the next steps. Thanks!

**Comment by iwiznia** [comment]
I think we need to make this internal because it seems the server is not sending the correct pusher events

**Comment by iwiznia** [comment]
lol it was already marked as internal.... not sure why we were doing proposals

**Comment by alexpensify** [comment]
I'll need to chat in the BZ room and figure out if I assign `engineering` or another label.

**Comment by alexpensify** [comment]
No update

**Comment by alexpensify** [comment]
I've asked for feedback here: 

https://expensify.slack.com/archives/C01SKUP7QR0/p1698338463683009

**Comment by iwiznia** [comment]
Looking into this I think that the bug is old and does not apply anymore?
Like it seems we don't display the name in the LHN ever, I assume that's the intended purpose.
@alexpensify if you agree, please close this.
